app/routes.js

```diff
@@ -40,19 +42,26 @@ export const requireAuth = (api) => (nextState, replace) => {
  *
  * @return {boolean|null} returns true if hash mapping happened
  */
-export const requireAuthAndNoPatient = (api) => (nextState, replace, cb) => {
-  if (!api.user.isAuthenticated()) {
+export const requireAuthAndNoPatient = (api, store) => (nextState, replace, cb) => {
+  let { blip: state } = store.getState();
+
+  if (!state.isLoggedIn && !api.user.isAuthenticated()) {
     replace('/login');
     return cb();
-  }
-  else {
-    api.user.get(function(err, user) {
-      if (personUtils.isPatient(user)) {
+  } else {
+    if (state.loggedInUser) {
+      if (personUtils.isPatient(state.loggedInUser)) {
```

Comment:
Yea, full user currently. In future we'll have to refactor all uses of `loggedInUser` so I think it is ok to leave this as is when we merge this branch, and then deal with normalisation task as a separate thing.